forge-game/src/main/java/forge/game/ability/effects/MillEffect.java

```diff
@@ -37,51 +33,31 @@ public void resolve(SpellAbility sa) {
             destination = ZoneType.Graveyard;
         }
 
-        Map<AbilityKey, Object> moveParams = AbilityKey.newMap();
-        final CardZoneTable table = AbilityKey.addCardZoneTableParams(moveParams, sa);
-
-        for (final Player p : getTargetPlayers(sa)) {
-            if (!p.isInGame()) {
-                continue;
-            }
+        final PlayerCollection millers = getTargetPlayers(sa);
 
-            String toZoneStr = destination.equals(ZoneType.Graveyard) ? "" : " (" +
-                    Localizer.getInstance().getMessage("lblMilledToZone", destination.getTranslatedName()) + ")";
-            if (sa.hasParam("Optional")) {
+        if (sa.hasParam("Optional")) {
+            final PlayerCollection toRemove = new PlayerCollection();
+            for (Player p : millers) {
                 String d = destination.equals(ZoneType.Graveyard) ? "" : " (" + destination.getTranslatedName() + ")";
                 final String prompt = TextUtil.concatWithSpace(Localizer.getInstance().
                         getMessage("lblDoYouWantToMill", Lang.nounWithNumeral(numCards, "card"), d));
                 // CR 701.13b
                 if (numCards > p.getZone(ZoneType.Library).size() || !p.getController().confirmAction(sa, null, prompt, null)) {
-                    continue;
-                }
-            }
-            final CardCollectionView milled = p.mill(numCards, destination, sa, moveParams);
-            // Reveal the milled cards, so players don't have to manually inspect the
-            // graveyard to figure out which ones were milled.
-            if (!facedown && reveal) { // do not reveal when exiling face down
-                if (showRevealDialog) {
-                    final String message = Localizer.getInstance().getMessage("lblMilledCards");
-                    final boolean addSuffix = !toZoneStr.equals("");
-                    game.getAction().reveal(milled, destination, p, false, message, addSuffix);
-                }
-                p.getGame().getGameLog().add(GameLogEntryType.ZONE_CHANGE, p + " milled " +
-                        Lang.joinHomogenous(milled) + toZoneStr + ".");
-            }
-            if (destination.equals(ZoneType.Exile)) {
-                for (final Card c : milled) {
-                    handleExiledWith(c, sa);
-                    if (facedown) {
-                        c.turnFaceDown(true);
-                    }
```

Comment:
the Mill into Exile got reworked into a better Exile effect, right?

Reply:
yea that was old code when Mill was used wrongly anyway